packages/@extension-kid/notification-service/NotificationService.ts

```diff
@@ -0,0 +1,48 @@
+import { injectable, inject, named } from "inversify";
+import { ExtensionProvider } from "@extension-kid/core";
+
+import { Notification } from "./Notification";
+
+const NotificationServiceExtensionType = Symbol("NotificationServiceExtension");
+const NotificationServiceType = Symbol("NotificationService");
+interface NotificationServiceExtensionInterface {
+  id: symbol;
+  supportedNotifications(): symbol[];
+  push(notification: Notification): void;
+}
+
+@injectable()
+class NotificationService {
+  extensionType = NotificationServiceExtensionType;
+  _extensionProvider: ExtensionProvider<NotificationServiceExtensionInterface>;
+
+  constructor(
+    @inject(ExtensionProvider)
+    @named(NotificationServiceExtensionType)
+    extensionProvider: ExtensionProvider<NotificationServiceExtensionInterface>
+  ) {
+    this._extensionProvider = extensionProvider;
+  }
+
+  push(notification: Notification): void {
+    let filteredExtensions: NotificationServiceExtensionInterface[] = [];
+    this._extensionProvider
+      .getAllExtensions()
+      .reduce((extensions, extension) => {
+        if (extension.supportedNotifications().includes(notification.type)) {
+          extensions.push(extension);
```

Comment:
curious why you decided to ask an extension instead of fanning out the notifications and let extension decide what to do?

Reply:
Could we use `filter` instead of `reduce`? Having `filteredExtensions` looks unnecessary to me.